### PR TITLE
PROD-73: Application layouts + Bootstrap + Haml

### PR DESCRIPTION
**Pull request by gnaizer**
# Reviewers

- [x] @vikabelous
- [x] @okohuch 
- [x] @Pavliv
- [x] @imysheniuk

# Jira task

http://ssu-jira.softserveinc.com/browse/PROD-73

# Description

Create application layout

# Checklist:

- [x] Rubocop doesn't raise any warnings and offenses
screenshot from 2018-04-09 16-08-14

- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

File: app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -2,39 +2,68 @@
 %html
   %head
     %meta{content: "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}
-    %title ProDobro
+    %title BootstrapApp
```
Comment (okohuch) [inline app/views/layouts/application.html.haml]: `BootstrapApp` instead of `ProDobro`?

**Comment by okohuch** [comment]
@gnaizer Do rebase and fix conflicts. Check review boxes. Also add a screen shot of the application.

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+.footer-copyright a {
+  color: #fff;
+  text-decoration: none;
+}
+
+.footer-copyright a:hover {
+  color: #0AB0C0;
+  text-decoration: none;
+}
```
Comment (imysheniuk) [inline app/assets/stylesheets/application.scss]: Please, combine "footer-copyright" styles into one

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+.list-unstyled p a {
+  color: #fff;
+  text-decoration: none;
+}
+
+.list-unstyled p a:hover {
+  color: #0AB0C0;
+}
```
Comment (imysheniuk) [inline app/assets/stylesheets/application.scss]: Please, combine "list-unstyled" styles into one

File: app/assets/stylesheets/application.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+.dropdown-menu a:hover {
+  background-color: #0AB0C0;
+}
+
+.page-footer {
+  color: #fff;
+}
+
+.list-unstyled p a {
+  color: #fff;
```
Comment (imysheniuk) [inline app/assets/stylesheets/application.scss]: if you use color more than ones, use variable

**Comment by gnaizer** [comment]
screenshot from 2018-04-10 14-07-17

File: db/schema.rb [hunk 5]
```diff
@@ -11,29 +11,30 @@
...
+    t.string "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string "unconfirmed_email"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["confirmation_token"], name: "index_users_on_confirmation_token", unique: true
+    t.index ["email"], name: "index_users_on_email", unique: true
+    t.index ["reset_password_token"], name: "index_users_on_reset_password_token", unique: true
   end
```
Comment (imysheniuk) [inline db/schema.rb]: Do you need this space?

**Review by imysheniuk — APPROVED** [review]

**Review by Pavliv — APPROVED** [review]

File: app/views/layouts/application.html.haml [hunk 6]
```diff
@@ -0,0 +1,70 @@
+!!!
+%html
+  %head
+    %meta{content: "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}
+    %title ProDobro
+    = csrf_meta_tags
+    = stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload'
+    = javascript_include_tag 'application', 'data-turbolinks-track': 'reload'
+    %link{href: "/images/hand.png", rel: "shortcut icon", type: "image/x-icon"}
```
Comment (viktoriia-bilous) [inline app/views/layouts/application.html.haml]: Please use [asset tag helpers](http://guides.rubyonrails.org/layouts_and_rendering.html#asset-tag-helpers) for assets

File: app/views/layouts/application.html.haml [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    %link{crossorigin: "anonymous", href: "https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css", integrity: "sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm", rel: "stylesheet"}
+    %script{crossorigin: "anonymous", defer: "defer", integrity: "sha384-8iPTk2s/jMVj81dnzb/iFR2sdA7u06vHJyyLlAd4snFpCl/SnyUjRrbdJsw1pGIl", src: "https://use.fontawesome.com/releases/v5.0.9/js/all.js"}
+  %body
+    %nav.navbar.navbar-expand-lg.navbar-dark.bg-dark.fixed-top
+      %a.navbar-brand{href: "#"}
+        %img{alt: "ProDobro", src: "/assets/prodobro2-9d9dcdf002f446dd3b77ccea0091ee1f75713ad69bdc8288e23a41834dfe29c2.png"}
```
Comment (viktoriia-bilous) [inline app/views/layouts/application.html.haml]: Same here

File: app/views/layouts/application.html.haml [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      %button.navbar-toggler{"aria-controls" => "navbarSupportedContent", "aria-expanded" => "false", "aria-label" => "Toggle navigation", "data-target" => "#navbarSupportedContent", "data-toggle" => "collapse", type: "button"}
+        %span.navbar-toggler-icon
+      #navbarSupportedContent.collapse.navbar-collapse
+        %ul.navbar-nav.ml-auto
+          %li.nav-item.active
+            %a.nav-link{href: "#"} News
```
Comment (viktoriia-bilous) [inline app/views/layouts/application.html.haml]: I18n

File: app/views/layouts/application.html.haml [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+              %i.fa.fa-envelope.mr-3
+              info@prodobro.com
+            %p
+              %i.fa.fa-phone.mr-3
+              + 01 234 567 88
+            %p
+              %i.fa.fa-print.mr-3
+              + 01 234 567 89
+      .footer-copyright.py-3.text-center
+        © 2018 Copyright:
```
Comment (viktoriia-bilous) [inline app/views/layouts/application.html.haml]: Year should not be hardcoded

**Comment by okohuch** [comment]
@gnaizer any updates?

File: app/assets/stylesheets/application.scss [hunk 10]
```diff
@@ -1,3 +1,92 @@
 @import "bootstrap";
 @import "variables";
 @import "users";
+
+$color-blue: #0AB0C0;
```
Comment (viktoriia-bilous) [inline app/assets/stylesheets/application.scss]: Please move to **variables.scss**

File: app/assets/stylesheets/application.scss [hunk 11]
```diff
@@ -1,3 +1,92 @@
...
+
+$color-white: #fff;
```
Comment (viktoriia-bilous) [inline app/assets/stylesheets/application.scss]: Same here

File: app/assets/stylesheets/users.scss [hunk 12]
```diff
@@ -2,8 +2,7 @@ a {
   color: $turquoise;
   &:hover {
     text-decoration: none;
-    color: $turquoise;
-    font-weight: bold;
+    color: #1c4e60;
```
Comment (viktoriia-bilous) [inline app/assets/stylesheets/users.scss]: Define as variable

File: app/views/layouts/application.html.haml [hunk 13]
```diff
@@ -0,0 +1,93 @@
+!!!
+%html
+  %head
+    %meta{content: "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}
+    %title ProDobro
```
Comment (okohuch) [inline app/views/layouts/application.html.haml]: change to use I18n everywhere

**Review by viktoriia-bilous — APPROVED** [review]

**Review by okohuch — APPROVED** [review]

**Pull request merged**